plugins/outputs/groundwork/groundwork.go

```diff
@@ -0,0 +1,296 @@
+package groundwork
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"strconv"
+
+	"github.com/gwos/tcg/sdk/clients"
+	"github.com/gwos/tcg/sdk/transit"
+	"github.com/hashicorp/go-uuid"
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/outputs"
+)
+
+// Login and logout routes from Groundwork API
+const (
+	loginURL  = "/api/auth/login"
+	logoutURL = "/api/auth/logout"
+)
+
+const (
+	sampleConfig = `
+  ## URL of your groundwork instance.
+  url = "https://groundwork.example.com"
+
+  ## Agent uuid for GroundWork API Server.
+  agent_id = ""
+
+  ## Username and password to access GroundWork API.
+  username = ""
+  password = ""
+
+  ## Default display name for the host with services(metrics).
+  # default_host = "telegraf"
+
+  ## Default service state.
+  # default_service_state = "SERVICE_OK"
+
+  ## The name of the tag that contains the hostname.
+  # resource_tag = "host"
+`
+)
+
+type Groundwork struct {
+	Server              string           `toml:"url"`
+	AgentID             string           `toml:"agent_id"`
+	Username            string           `toml:"username"`
+	Password            string           `toml:"password"`
+	DefaultHost         string           `toml:"default_host"`
+	DefaultServiceState string           `toml:"default_service_state"`
+	ResourceTag         string           `toml:"resource_tag"`
+	Log                 telegraf.Logger  `toml:"-"`
+	GWClient            clients.GWClient `toml:"-"`
```

Comment:
I think you don't need to export this field, do you?
```suggestion
	client            clients.GWClient
```

Reply:
Done